frontend/src/core/comments/components/AddComment.js

```diff
@@ -25,69 +34,91 @@ export default function AddComments(props: Props) {
         addComment,
     } = props;
 
-    const commentInput: any = React.useRef();
-    const minRows = 1;
-    const maxRows = 6;
+    let [editorState, setEditorState] = React.useState(
+        EditorState.createEmpty()
+    )
+    const editor: any = React.useRef(null);
+
+    React.useEffect(() => {
+        focusEditor()
+    }, []);
 
     if (!user) {
         return null;
     }
+    
+    const onChange = (editorState) => {
+        setEditorState(editorState);
+    }
 
-    const handleOnChange = () => {
-        const textAreaLineHeight = 24;
-        commentInput.current.rows = minRows;
-
-        const currentRows = Math.trunc(commentInput.current.scrollHeight / textAreaLineHeight);
+    function focusEditor() {
+        editor.current.focus();
+    }
 
-        if (currentRows < maxRows) {
-            commentInput.current.rows = currentRows;
-        }
-        else {
-            commentInput.current.rows = maxRows;
+    function keyBindingFn(e: SyntheticKeyboardEvent<>) {
+        if (e.keyCode === 13 && e.shiftKey === false) {
+            return 'submit-on-enter'; 
         }
+        
+        return getDefaultKeyBinding(e)
     }
 
-    const handleOnKeyDown = (event: SyntheticKeyboardEvent<>) => {
-        if (event.keyCode === 13 && event.shiftKey === false) {
-            submitComment(event);
+    const handleKeyCommand = (command: string) => {
+        if (command === 'submit-on-enter') {
+            submitComment();
+            return 'handled';
         }
+        return 'not-handled';
     }
 
-    const submitComment = (event: SyntheticEvent<>) => {
-        event.preventDefault();
-        const comment = commentInput.current.value;
+    const clearEditorContent = () => {
+        let contentState = editorState.getCurrentContent();
+        const firstBlock = contentState.getFirstBlock();
+        const lastBlock = contentState.getLastBlock();
+        const allSelected = new SelectionState({
+            anchorKey: firstBlock.getKey(),
+            anchorOffset: 0,
+            focusKey: lastBlock.getKey(),
+            focusOffset: lastBlock.getLength(),
+            hasFocus: true
+        });
+        contentState = Modifier.removeRange(contentState, allSelected, 'backward');
+        editorState = EditorState.push(editorState, contentState, 'remove-range');
```

Comment:
Unless I missed something, this is replacing the `editorState` that you defined with `useState` outside this function. I assume that's not what you intended to do, especially considering you're using `setEditorState` right below. I'd recommend using a different variable name here, or at least declaring it with `let` to make sure it's not the same as in the component's scope.

Reply:
This method seems pretty cumbersome to just clear the content. Can we instead use something like this?
https://stackoverflow.com/questions/37463757/how-to-clear-input-field-in-draft-js/39029383#39029383

I've also found clearEditorContent in https://github.com/jpuri/draftjs-utils.

Reply:
Sadly, there is no easy way to reset the state and clear the editor. This solution was from an [issue submitted](https://github.com/facebook/draft-js/issues/445) about this on the draft.js repo. (I had to correct line 87 to be `...forceSelection(**editorState**, ....) as the solution given did not work as written and was admittedly (via the disclaimer) not tested prior to posting)

I did see that the issue post mentioned above had `let { editorState } = this.state;` , but when I tried to adjust given that we are using hooks it did not work. Also, giving this a different variable name does not work either. In essence, neither of those options clears the contents and allows for a new comment to be posted.

If you have an alternate way of approaching this, I would be more than happy to pursue it.

Reply:
@mathjazz I tried that first solution with no luck, and I'm pretty sure I also tried the second solution (drafjs-utils), also with no luck, but will try again as I can't quite remember (I tried MANY options for this problem... 😄 )

Reply:
Maybe use [createFromText()](https://draftjs.org/docs/api-reference-content-state/#createfromtext)?

```
editorState = EditorState.createWithContent(ContentState.createFromText(''));
setEditorState(editorState);
```

Reply:
@mathjazz   `createFromText()` introduced 2 new bugs: 
1. the comment now renders on top of the existing comments instead of the bottom.
2. The editor exhibits weird behaviors on the next entry (cursor moves in front of the first char entered and the ability to delete text is gone as well as possible other issues)

Also, after taking another look at `draftjs-utils` it appears that `clearrEditorContent` is essentially the same implementation that I have (I still can't remember if added and tested it out, but I feel like I did.  Either way, I will still give it a go, but wanted you to be aware that while it would cleaner on the surface it would essentially be the same 'under the hood')